### PR TITLE
QueryBuilder: Handle user-entry semantic labels on submit

### PR DESCRIPTION
**Pull request by lukebjerring**
## Description
Handles an edge-case where the user enters a semantic label (e.g. `stable`) manually as a label, rather than clicking the dropdowns 4 times.

## Review Information
As per https://github.com/web-platform-tests/wpt.fyi/pull/769#issuecomment-440844710 
- Visit /runs
- Edit the query, type a semantic label (e.g. a channel, like `stable`)
- Submit, see the expected results
- Edit the query, type a different semantic label (e.g. `experimental`)

### CODE REVIEW

File: webapp/components/test-runs-query-builder.html [hunk 1]
```diff
@@ -181,6 +179,22 @@ <h3>
...
 
+      handleSubmit() {
+        // Handle the edge-case that the user typed a label for channel or source, etc.
+        const productBuilders = this.shadowRoot.querySelectorAll('product-builder');
+        for (const semantic of window.wpt.SemanticLabels) {
+          for (const label of semantic.values) {
+            if (this.labels.includes(label)) {
+              this.labels = this.labels.filter(l => l !== label);
+              for (const p of productBuilders) {
+                p[semantic.property] = label;
```
Comment (foolip) [inline webapp/components/test-runs-query-builder.html]: Nice, a pretty deep loop, but all makes sense without comments to explain it :)
